packages/server/utils/analytics/amplitude/AmplitudeAnalytics.ts

```diff
@@ -0,0 +1,29 @@
+import {init, track} from '@amplitude/analytics-node'
+import {AnalyticsEvent} from '../analytics'
+import {getUserById} from '../../../postgres/queries/getUsersByIds'
+import PROD from '../../../PROD'
+
+const {AMPLITUDE_WRITE_KEY} = process.env
+
+/**
+ * Wrapper for amplitude providing a more typesafe interface
+ */
+export class AmplitudeAnalytics {
+  constructor() {
+    init(AMPLITUDE_WRITE_KEY || 'x', {
+      flushQueueSize: PROD ? 20 : 1,
+      optOut: !AMPLITUDE_WRITE_KEY
+    })
+  }
+
+  async track(userId: string, event: AnalyticsEvent, properties?: Record<string, any>) {
+    // used as a failsafe for PPMIs
+    if (!AMPLITUDE_WRITE_KEY) return
+    const user = await getUserById(userId)
```

Comment:
If `email` is required, we should probably use `dataloader` here instead of hitting PQ for every single event we track

Reply:
That seems like a good idea.  I'll make that change!

Reply:
Yes, please make that change